libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.service.ts

```diff
@@ -46,6 +51,40 @@ export class NationalRegistryXRoadService {
     }
   }
 
+  async getNationalRegistryResidenceHistory(
+    nationalId: string,
+    authToken: string,
+  ): Promise<NationalRegistryResidenceHistory> {
+    const historyList = await this.nationalRegistryFetch<Array<Heimili>>(
+      `/${nationalId}/buseta`,
+      authToken,
+    )
+
+    const history = historyList.map((heimili: Heimili) => {
+      // API says Date, but is string -- fallback in case that changes in the future
+      const date =
+        typeof heimili.breytt === 'string'
+          ? new Date(heimili.breytt)
+          : heimili.breytt
```

Comment:
Won't this always be sent as a string, since the api stringifies the values before sending them? 

If so then the typeof check is not needed.

Reply:
I might be doing something wrong, but as far as I could tell, this is a bug or at the very least a very weird feature.

The type is registered as a Date according to the openapi => typescript client generator. however, the data is indeed a string, so, eg. calling anything such as .getTime() on it will always fail, even though the typescript checker does not complain.

I didn't dig deep enough to figure out whether the bug is in the generator, in the deserializer or what, but this is what I was observing

Reply:
number? if that would be the case, then we have bigger problems. in that case, as the code is written, the value of it would be a number when calling getTime() etc on it, which would result in an error/exception, and that would be discoverable.

however, if you put a number into the date constructor, you will get a date.. but since we don't really know what the format of that number would be (Date-compatible? unix timestamp w/o milliseconds? something else?) .. it would mean we'd just introduce a bug/wrong information, resulting in bad data that probably wouldn't be discovered until it is reported by a user